astropy/time/formats.py

```diff
@@ -72,6 +73,11 @@ def _regexify_subfmts(subfmts):
     return tuple(new_subfmts)
 
 
+class TimeMaskedArray(np.ma.MaskedArray):
```

Comment:
Why do we need a new class? It seems you want to avoid `repr`, but where is that a problem?